clients/src/main/java/org/apache/kafka/common/requests/ControlledShutdownRequest.java

```diff
@@ -16,33 +16,53 @@
 import org.apache.kafka.common.protocol.ApiKeys;
 import org.apache.kafka.common.protocol.Errors;
 import org.apache.kafka.common.protocol.ProtoUtils;
-import org.apache.kafka.common.protocol.types.Schema;
 import org.apache.kafka.common.protocol.types.Struct;
 
 import java.nio.ByteBuffer;
 import java.util.Collections;
 
 public class ControlledShutdownRequest extends AbstractRequest {
+    private static final String BROKER_ID_KEY_NAME = "broker_id";
 
-    private static final Schema CURRENT_SCHEMA = ProtoUtils.currentRequestSchema(ApiKeys.CONTROLLED_SHUTDOWN_KEY.id);
+    public static class Builder extends AbstractRequest.Builder<ControlledShutdownRequest> {
+        private int brokerId;
```

Comment:
Seems this field could be final. Might be worth a pass over the other request types to locate other fields that could be final.